Client/Coordinators/Router/Route.swift

```diff
@@ -46,26 +66,175 @@ enum Route {
+                    let tab = tabs[uuid]
+                    self = .search(url: tab?.url, tabId: uuid)
+                } else {
+                    self = .search(url: nil, isPrivate: false)
+                }
+
+            case .widgetTabsLargeOpenUrl:
+                // Widget Tabs Quick View - large
+                let tabs = SimpleTab.getSimpleTabs()
+                if let uuid = urlScanner.value(query: "uuid"), !tabs.isEmpty {
+                    let tab = tabs[uuid]
+                    self = .search(url: tab?.url, tabId: uuid)
+                } else {
+                    self = .search(url: nil, isPrivate: false)
+                }
+
+            case .fxaSignIn:
+                return nil
+            }
+
+            recordTelemetry(input: host, isPrivate: isPrivate)
+        } else if urlScanner.isHTTPScheme {
+            TelemetryWrapper.gleanRecordEvent(category: .action, method: .open, object: .asDefaultBrowser)
+            RatingPromptManager.isBrowserDefault = true
+            // Use the last browsing mode the user was in
+            let isPrivate = UserDefaults.standard.bool(forKey: PrefsKeys.LastSessionWasPrivate)
+            self = .search(url: url, isPrivate: isPrivate)
+        } else {
+            return nil
+        }
+    }
+
+    func recordTelemetry(input: DeeplinkInput.Host, isPrivate: Bool) {
+        switch input {
+        case .deepLink, .fxaSignIn, .openUrl, .openText, .glean:
+            return
+        case .widgetMediumTopSitesOpenUrl:
+            TelemetryWrapper.recordEvent(category: .action, method: .open, object: .mediumTopSitesWidget)
+        case .widgetSmallQuickLinkOpenUrl:
+            TelemetryWrapper.recordEvent(category: .action, method: .open, object: .smallQuickActionSearch)
+        case .widgetMediumQuickLinkOpenUrl:
+            TelemetryWrapper.recordEvent(category: .action, method: .open, object: isPrivate ?.mediumQuickActionPrivateSearch:.mediumQuickActionSearch)
+        case .widgetSmallQuickLinkOpenCopied:
+            TelemetryWrapper.recordEvent(category: .action, method: .open, object: .smallQuickActionClosePrivate)
+        case .widgetMediumQuickLinkOpenCopied:
+            TelemetryWrapper.recordEvent(category: .action, method: .open, object: .mediumQuickActionClosePrivate)
+        case .widgetSmallQuickLinkClosePrivateTabs:
+            TelemetryWrapper.recordEvent(category: .action, method: .open, object: .smallQuickActionClosePrivate)
+        case .widgetMediumQuickLinkClosePrivateTabs:
+            TelemetryWrapper.recordEvent(category: .action, method: .open, object: .mediumQuickActionClosePrivate)
+        case .widgetTabsMediumOpenUrl:
+            TelemetryWrapper.recordEvent(category: .action, method: .open, object: .mediumTabsOpenUrl)
+        case .widgetTabsLargeOpenUrl:
+            TelemetryWrapper.recordEvent(category: .action, method: .open, object: .largeTabsOpenUrl)
+        }
+    }
+}
+
+public extension URL {
```

Comment:
See comment about extensions from classes we don't own [here](https://mozilla.slack.com/archives/C04TSDUEQPP/p1680788962206709?thread_ts=1680787030.391509&cid=C04TSDUEQPP)